bodo/io/iceberg/monkey_patch.py

```diff
@@ -0,0 +1,79 @@
+import pyiceberg.io
```

Comment:
Sadly I needed to monkey patch a few things from PyIceberg to fix various issues. See below

Reply:
Makes sense, I'm going to leave it like this for the time being because I will have to make some large changes when PyIceberg 0.9 comes out anyways